### PR TITLE
[UPI] Show error instead of disabled button

### PR DESCRIPTION
**Pull request by goergisn**
## Summary
- Showing an error item when the submit button was tapped without having an app selected

### TODOs

- [x] Localization

### CODE REVIEW

File: AdyenComponents/UPI/UPIComponent.swift [hunk 1]
```diff
@@ -330,17 +350,26 @@ private extension UPIComponent {
...
+    private func showError() {
+        errorItem.isHidden.wrappedValue = false
+        UIAccessibility.post(
+            notification: .announcement,
+            argument: "\(localizedString(.errorTitle, configuration.localizationParameters)): \(errorItem.message ?? "")"
+        )
+    }
+
+    private func hideError() {
+        errorItem.message = nil
```
Comment (erenbesel) [inline AdyenComponents/UPI/UPIComponent.swift]: If set to nill and need to re-show, message will be gone

File: AdyenComponents/UPI/UPIComponent.swift [hunk 2]
```diff
@@ -330,17 +350,26 @@ private extension UPIComponent {
...
+        errorItem.isHidden.wrappedValue = true
+    }
+    
     func canSubmit() -> Bool {
```
Comment (erenbesel) [inline AdyenComponents/UPI/UPIComponent.swift:373]: Many functions seem to be missing `internal` keyword or github messing up?

**Review by erenbesel — APPROVED** [review]

File: AdyenComponents/UPI/UPIComponent.swift (hunk 1 above)
Comment (goergisn) [reply AdyenComponents/UPI/UPIComponent.swift]: Ohh good catch! will adjust the tests to catch it

File: AdyenComponents/UPI/UPIComponent.swift (hunk 2 above)
Comment (goergisn) [reply AdyenComponents/UPI/UPIComponent.swift:373]: We're in a private extension here - removed the other keywords

File: Adyen/UI/Form/Items/Error/FormErrorItemView.swift [hunk 3]
```diff
@@ -19,7 +19,7 @@ internal final class FormErrorItemView: FormItemView<FormErrorItem> {
         bind(item.$message, to: self, at: \.accessibilityLabel)
         isHidden = item.isHidden.wrappedValue
         addSubview(containerView)
-        containerView.adyen.anchor(inside: layoutMarginsGuide, with: UIEdgeInsets(top: 16, left: 16, bottom: -16, right: -16))
+        containerView.adyen.anchor(inside: layoutMarginsGuide, with: UIEdgeInsets(top: 8, left: 0, bottom: -8, right: 0))
```
Comment (atmamont) [inline Adyen/UI/Form/Items/Error/FormErrorItemView.swift:22]: It would be great to see all UI changes in the code supported with screenshots

**Review by atmamont — APPROVED** [review]

File: AdyenComponents/UPI/UPIComponent.swift [hunk 4]
```diff
@@ -82,6 +82,10 @@ public final class UPIComponent: PaymentComponent,
         self.upiPaymentMethod = paymentMethod
         self.context = context
         self.configuration = configuration
+        
+        if upiAppsList.isEmpty {
```
Comment (atmamont) [inline AdyenComponents/UPI/UPIComponent.swift:86]: Let's add a test for this right away?

File: Adyen/UI/Form/Items/Error/FormErrorItemView.swift (hunk 3 above)
Comment (goergisn) [reply Adyen/UI/Form/Items/Error/FormErrorItemView.swift:22]: Will be once CI works again (Need to run snapshot updates)

**Comment by atmamont** [comment]
> # ✅ No changes detected
> _Comparing `/Users/runner/work/adyen-ios/adyen-ios` to `https://github.com/Adyen/adyen-ios.git @ UPI_intent_base`_
> 
> **Analyzed modules:** Adyen, AdyenActions, AdyenCard, AdyenCashAppPay, AdyenComponents, AdyenDelegatedAuthentication, AdyenDropIn, AdyenEncryption, AdyenSession, AdyenSwiftUI, AdyenTwint, AdyenWeChatPay

This is so concise, informative and reassuring we didn't break anything :) 👏

**Pull request merged**